File: LAB3/app/src/main/java/com/example/tipper/ChartActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.github.mikephil.charting.charts.BarChart;
import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.data.BarData;
import com.github.mikephil.charting.data.BarDataSet;
import com.github.mikephil.charting.data.BarEntry;
import com.github.mikephil.charting.formatter.IAxisValueFormatter;
import com.github.mikephil.charting.utils.ColorTemplate;

import java.util.ArrayList;

public class ChartActivity extends AppCompatActivity {

    BarChart barChart;
    Button btnBack;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_chart);
        barChart = (BarChart) findViewById(R.id.barchart);

        barChart.setDrawBarShadow(false);
        barChart.setDrawValueAboveBar(true);
        barChart.setMaxVisibleValueCount(50);
        barChart.setPinchZoom(false);
        barChart.setDrawGridBackground(true);

        ArrayList<BarEntry> barEntries = new ArrayList<>();

        barEntries.add(new BarEntry(1, 25.66f));
        barEntries.add(new BarEntry(2, 25.06f));
        barEntries.add(new BarEntry(3, 24.15f));
        barEntries.add(new BarEntry(4, 23.94f));

        BarDataSet barDataSet = new BarDataSet(barEntries, "Date Set");
        barDataSet.setColors(ColorTemplate.COLORFUL_COLORS);

        BarData data = new BarData(barDataSet);
        data.setBarWidth(0.7f);

        barChart.setData(data);

        String[] months = new String[] {"Jan","Fab", "Mar", "April", "May", "Jun"};
        XAxis xAxis = barChart.getXAxis();
        xAxis.setValueFormatter(new valueFormat(months));
        xAxis.setPosition(XAxis.XAxisPosition.BOTH_SIDED);
        xAxis.setGranularity(1);
        xAxis.setCenterAxisLabels(true);
        xAxis.setAxisMinimum(1);

        btnBack = (Button) findViewById(R.id.btnBack);
        btnBack.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(ChartActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });
    }
    public class valueFormat implements IAxisValueFormatter {

        private String[] values;

        public valueFormat(String[] val) {
            this.values = val;
        }

        @Override
        public String getFormattedValue(float value, AxisBase axis) {
            return values[(int)value];
        }
    }
}

File: LAB4/app/src/main/java/com/example/myapplicationpamo/ChartActivity.kt
package com.example.myapplicationpamo

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.github.mikephil.charting.data.BarData
import com.github.mikephil.charting.data.BarDataSet
import com.github.mikephil.charting.data.BarEntry
import kotlinx.android.synthetic.main.activity_chart.*

class ChartActivity : AppCompatActivity() {

    lateinit var barList:ArrayList<BarEntry>
    lateinit var barDataSet: BarDataSet
    lateinit var barData: BarData
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_chart)

        barList= ArrayList()
        barList.add(BarEntry(1f,25.66f))
        barList.add(BarEntry(2f,25.06f))
        barList.add(BarEntry(3f,24.15f))
        barList.add(BarEntry(4f,23.94f))


        barDataSet= BarDataSet(barList,"months")
        barData= BarData(barDataSet)
        barchart.data = barData
        barDataSet.valueTextColor= Color.BLACK
        barDataSet.valueTextSize=10f
    }
}